AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AccountControllerTests.cs

```diff
@@ -802,39 +800,156 @@ public void ForgotPasswordGetHasAllowAnonymousAttribute()
+
+            var userManager = CreateUserManagerMock();
+
+            var user = new ApplicationUser();
+
+            userManager.Setup(x => x.FindByNameAsync(email)).Returns(() => Task.FromResult(user));
+            userManager.Setup(x => x.IsEmailConfirmedAsync(user)).Returns(() => Task.FromResult(true));
+            userManager.Setup(x => x.GeneratePasswordResetTokenAsync(user)).Returns(() => Task.FromResult(It.IsAny<string>()));
+
+            var emailSender = new Mock<IEmailSender>();
+            emailSender.Setup(x => x.SendEmailAsync(email, It.IsAny<string>(), It.IsAny<string>())).Returns(() => Task.FromResult(It.IsAny<Task>()));
+
+            var sut = new AccountController(userManager.Object, null, emailSender.Object, null, null);
+
+            var urlHelper = new Mock<IUrlHelper>();
+
+            sut.SetFakeHttpRequestSchemeTo(requestScheme);
+            sut.Url = urlHelper.Object;
+
+            await sut.ForgotPassword(vm);
+
+            urlHelper.Verify(mock => mock.Action(It.Is<UrlActionContext>(uac => 
+                        uac.Action == "ResetPassword" 
+                        && uac.Controller == "Account" 
+                        && uac.Protocol == requestScheme)),Times.Once);
         }
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task ForgotPasswordPostInvokesSendEmailAsyncWithCorrectParametersWhenModelStateIsValidAndUserIsNotNullAndUsersEmailHasBeenVerified()
         {
-            //delete this line when starting work on this unit test
-            await taskFromResultZero;
+            var email = "user@domain.tld";
+            const string callbackUrl = "callbackUrl";
+
+            var vm = new ForgotPasswordViewModel()
+            {
+                Email = email
+            };
+
+            var userManager = CreateUserManagerMock();
+
+            var user = new ApplicationUser();
+
+            userManager.Setup(x => x.FindByNameAsync(email)).Returns(() => Task.FromResult(user));
+            userManager.Setup(x => x.IsEmailConfirmedAsync(user)).Returns(() => Task.FromResult(true));
+            userManager.Setup(x => x.GeneratePasswordResetTokenAsync(user)).Returns(() => Task.FromResult(It.IsAny<string>()));
+
+            var emailSender = new Mock<IEmailSender>();
+            emailSender.Setup(x => x.SendEmailAsync(email, It.IsAny<string>(), It.IsAny<string>())).Returns(() => Task.FromResult(It.IsAny<Task>()));
+
+            var sut = new AccountController(userManager.Object, null, emailSender.Object, null, null);
+
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper.Setup(x => x.Action(It.IsAny<UrlActionContext>())).Returns(callbackUrl);
+
+            sut.SetFakeHttpRequestSchemeTo(It.IsAny<string>());
+            sut.Url = urlHelper.Object;
+
+            await sut.ForgotPassword(vm);
+
+            emailSender.Verify(x => x.SendEmailAsync(email, It.IsAny<string>(), It.Is<string>(y => y.Contains(callbackUrl))), Times.Once);
```

Comment:
we want to be explicit with this test for all the values passed to `SendEmailAsync`. Right now, we're just testing that the third parameter contains the `callbackUrl`. I'd re-write as this:
`emailSender.Verify(x => x.SendEmailAsync(email, "Reset allReady Password", $"Please reset your allReady password by clicking here: <a href=\"{callbackUrl}\">link</a>"), Times.Once);`